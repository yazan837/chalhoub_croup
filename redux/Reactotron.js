import Reactotron from 'reactotron-react-native';
import {reactotronRedux} from 'reactotron-redux';

export default Reactotron.configure({name: 'chalhoub'})
  .useReactNative({
    asyncStorage: false,
    networking: {
      ignoreUrls: /symbolicate/,
    },
    editor: false,
    errors: {veto: stackFrame => false},
    overlay: false,
  })
  .use(reactotronRedux())
  .connect();
