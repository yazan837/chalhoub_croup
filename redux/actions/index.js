import home from './groups/home';

export default {
  ...home,
};
