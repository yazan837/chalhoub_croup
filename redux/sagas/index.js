import {fork} from 'redux-saga/effects';

import * as home from './groups/home';
const sagas = Object.values({
  ...home,
});

export default function* root() {
  for (let index = 0; index < sagas.length; index++) {
    yield fork(sagas[index]);
  }
}
