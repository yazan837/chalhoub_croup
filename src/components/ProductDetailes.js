import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  ScrollView,
  FlatList,
  StyleSheet,
} from 'react-native';
import {TextInput} from 'react-native-gesture-handler';
import {useSelector} from 'react-redux';
import {navigate} from '../../navigation/NavigationService';

const ProductDetailes = item => {
  const data = item.route.params.item;
  const allImages = useSelector(state => state.home.products);
  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.container}>
        <Image source={require('../../assets/Logo.png')} style={styles.logo} />
        <View style={styles.HeadLine}>
          <Text style={{fontSize: 16}}>MEN</Text>
          <Text style={{fontSize: 16}}>WOMEN</Text>
          <Text style={{fontSize: 16}}>KIDS</Text>
          <Text style={{fontSize: 16}}>FTW</Text>
          <Text style={{fontSize: 16}}>ACCESSORIES</Text>
        </View>
        <View style={styles.searchContainer}>
          <TextInput
            placeholder="search"
            style={{flex: 1, alignSelf: 'flex-start', width: '100%'}}
          />
        </View>
        <View style={styles.ImageContainer}>
          <Image source={{uri: data.image}} style={styles.productImage} />
          <TouchableOpacity style={styles.ResizeContainer}>
            <Image
              source={require('../../assets/resize.png')}
              style={styles.resizeImage}
            />
          </TouchableOpacity>
        </View>
        <View style={styles.TitleContainer}>
          <Text style={{fontSize: 20}}>{data.title}</Text>
          <Text style={{fontSize: 14}}>{data.price} $</Text>
        </View>
        <View style={styles.TitleContainer}>
          <Text style={{fontSize: 20, color: 'grey'}}>
            Avalibale Size : M L XL XXL
          </Text>
        </View>
        <View style={styles.TitleContainer}>
          <Text style={{fontSize: 20, color: 'grey'}}>COLORS</Text>
          <View style={styles.colorContainer}>
            <TouchableOpacity
              style={{borderWidth: 20, borderColor: 'red', borderRadius: 1}}
            />
            <TouchableOpacity
              style={{borderWidth: 20, borderColor: 'black', borderRadius: 1}}
            />
            <TouchableOpacity
              style={{borderWidth: 20, borderColor: 'blue', borderRadius: 1}}
            />
            <TouchableOpacity
              style={{borderWidth: 20, borderColor: 'green', borderRadius: 1}}
            />
          </View>
        </View>
        <View style={styles.boxContainer} />
        <View style={styles.TitleContainer}>
          <Text style={{fontSize: 20, color: 'grey', fontWeight: 'bold'}}>
            DESCRIPTION
          </Text>
          <Text style={styles.descriptionText}>{data.description}</Text>
        </View>
        <View style={{width: '95%', borderWidth: 0.6, marginTop: 10}} />
        <View style={styles.TitleContainer}>
          <Text
            style={{fontSize: 20, color: 'grey', fontWeight: 'bold'}}
            onPress={() => navigate('Home')}>
            SEE MORE PRODUCTS
          </Text>
          <FlatList
            data={allImages}
            horizontal={true}
            showsHorizontalScrollIndicator={false}
            renderItem={({item}) => {
              return (
                <View style={{margin: 10}}>
                  <Image
                    source={{uri: item.image}}
                    style={{width: 100, height: 200}}
                    resizeMode="center"
                  />
                </View>
              );
            }}
          />
        </View>
      </View>
    </ScrollView>
  );
};

export default ProductDetailes;

const styles = StyleSheet.create({
  container: {alignItems: 'center', backgroundColor: 'white'},
  logo: {width: 200, height: 200, resizeMode: 'center'},
  HeadLine: {
    flexDirection: 'row',
    width: '95%',
    justifyContent: 'space-between',
  },
  searchContainer: {
    alignItems: 'center',
    width: '95%',
    height: 40,
    borderWidth: 0.8,
    margin: 10,
  },
  ImageContainer: {
    borderWidth: 0.9,
    width: '95%',
    alignItems: 'center',
  },
  productImage: {width: 300, height: 400, resizeMode: 'center'},
  ResizeContainer: {alignSelf: 'flex-end', borderWidth: 0.5},
  resizeImage: {width: 25, height: 25, resizeMode: 'center'},
  TitleContainer: {width: '95%', alignItems: 'flex-start', marginTop: 20},
  colorContainer: {
    flexDirection: 'row',
    width: '50%',
    justifyContent: 'space-between',
  },
  boxContainer: {
    height: 30,
    borderRadius: 10,
    width: '95%',
    backgroundColor: 'grey',
    marginTop: 10,
  },
  descriptionText: {fontSize: 16, color: 'grey', marginTop: 10},
});
