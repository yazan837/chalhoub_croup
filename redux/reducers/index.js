import {combineReducers} from 'redux';

import home from './groups/home';

const appReducers = combineReducers({
  home,
});

export default (state, action) => {
  return appReducers(state, action);
};
