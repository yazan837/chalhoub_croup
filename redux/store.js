import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducers from './reducers';
import sagas from './sagas';

import Reactotron from './Reactotron';

const sagaMiddleware = createSagaMiddleware();
export default createStore(
  reducers,
  {},
  compose(applyMiddleware(sagaMiddleware), Reactotron.createEnhancer()),
);
sagaMiddleware.run(sagas);

//adb reverse tcp:9090 tcp:9090
//adb reverse tcp:8081 tcp:8081
