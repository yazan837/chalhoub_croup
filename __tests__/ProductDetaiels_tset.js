import React from 'react';
import {create} from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducers from '../redux/reducers';

const store = createStore(reducers);

import ProductDetailes from '../src/components/ProductDetailes';

jest.useFakeTimers();

const tree = create(
  <Provider store={store}>
    <ProductDetailes />
  </Provider>,
);

test('test ProductDetailes page ui', () => {
  expect(tree).toMatchSnapshot();
});
