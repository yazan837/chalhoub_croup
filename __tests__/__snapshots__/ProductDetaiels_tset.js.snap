// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProductDetailes page ui 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/Logo.png",
          }
        }
        style={
          Object {
            "height": 200,
            "resizeMode": "center",
            "width": 200,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "95%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          MEN
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          WOMEN
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          KIDS
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          FTW
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          ACCESSORIES
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderWidth": 0.8,
            "height": 40,
            "margin": 10,
            "width": "95%",
          }
        }
      >
        <TextInput
          collapsable={false}
          handlerTag={1}
          handlerType="NativeViewGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="search"
          style={
            Object {
              "alignSelf": "flex-start",
              "flex": 1,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderWidth": 0.9,
            "width": "95%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 400,
              "resizeMode": "center",
              "width": 300,
            }
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "borderWidth": 0.5,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/resize.png",
              }
            }
            style={
              Object {
                "height": 25,
                "resizeMode": "center",
                "width": 25,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
           $
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 20,
            }
          }
        >
          Avalibale Size : M L XL XXL
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 20,
            }
          }
        >
          COLORS
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "50%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "red",
                "borderRadius": 1,
                "borderWidth": 20,
                "opacity": 1,
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "black",
                "borderRadius": 1,
                "borderWidth": 20,
                "opacity": 1,
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 1,
                "borderWidth": 20,
                "opacity": 1,
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "green",
                "borderRadius": 1,
                "borderWidth": 20,
                "opacity": 1,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "borderRadius": 10,
            "height": 30,
            "marginTop": 10,
            "width": "95%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          DESCRIPTION
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "marginTop": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderWidth": 0.6,
            "marginTop": 10,
            "width": "95%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "grey",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          SEE MORE PRODUCTS
        </Text>
        <RCTScrollView
          data={Array []}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
